### PR TITLE
change the config-gateway format

### PR DESCRIPTION
**Pull request by dprotaso**
This puts the configuration in line with what we did in `net-istio` - details are here https://github.com/knative-extensions/net-istio/issues/1247

but tl;dr we changed the gateway config to take a list of gateways. This will allow us to do label filtering of gateways in the future without breaking the config format.

For now we restrict the number of gateways to a single one.

### CODE REVIEW

File: pkg/reconciler/ingress/config/gateway.go [hunk 1]
```diff
@@ -33,114 +32,128 @@ const (
...
-	for _, vis := range []v1alpha1.IngressVisibility{
-		v1alpha1.IngressVisibilityClusterLocal,
-		v1alpha1.IngressVisibilityExternalIP,
-	} {
-		if _, ok := visConfig[vis]; !ok {
-			return nil, fmt.Errorf("visibility %q must not be empty", vis)
+	if data, ok := cm.Data[externalGatewaysKey]; ok {
+		config.ExternalGateways, err = parseGatewayConfig(data)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to parse %q: %w", externalGatewaysKey, err)
```
Comment (izabelacg) [inline pkg/reconciler/ingress/config/gateway.go:99]: Some check is failing because error message is capitalized in some places

**Comment by dprotaso** [comment]
Reset PR to pull in latest fixes in the `main` branch

**Comment by dprotaso** [comment]
all good now 
@izabelacg PTAL again

/hold cancel

**Comment by dprotaso** [comment]
/override "Generate some code!"
/override "style / suggester / github_actions"

Workaround for prow merging being broken - https://github.com/knative/infra/issues/411

**Pull request merged**